apollo-router/src/uplink/license_enforcement.rs

```diff
@@ -197,14 +197,14 @@ impl LicenseEnforcementReport {
     }
 
     fn validate_schema(
-        schema: &apollo_compiler::ast::Document,
+        schema: &Schema,
         schema_restrictions: &Vec<SchemaRestriction>,
     ) -> Vec<SchemaViolation> {
         let link_specs = schema
-            .definitions
-            .iter()
-            .filter_map(|def| def.as_schema_definition())
-            .flat_map(|def| def.directives.get_all(LINK_DIRECTIVE_NAME))
+            .supergraph_schema()
+            .schema_definition
+            .directives
+            .get_all(LINK_DIRECTIVE_NAME)
             .filter_map(|link| {
                 ParsedLinkSpec::from_link_directive(link).map(|maybe_spec| {
```

Comment:
Should this be replaced with something from apollo-federation? (Through `schema.federation_supergraph().schema.metadata()`)